projects/valtimo/dossier-management/src/lib/components/dossier-management-connect-modal/dossier-management-connect-modal.component.ts

```diff
@@ -43,34 +44,49 @@ export class DossierManagementConnectModalComponent implements OnInit {
   public newDocumentProcessDefinitionStartableByUser = false;
   public processDocumentDefinitionExists: any = {};
 
+  private readonly _subscriptions = new Subscription();
+
   constructor(
-    private readonly translateService: TranslateService,
-    private readonly processService: ProcessService,
     private readonly documentService: DocumentService,
-    private readonly notificationService: NotificationService
+    private readonly dossierDetailService: DossierDetailService,
+    private readonly notificationService: NotificationService,
+    private readonly processService: ProcessService,
+    private readonly translateService: TranslateService
   ) {}
 
+  public ngOnDestroy(): void {
+    this._subscriptions.unsubscribe();
+  }
+
   public loadProcessDocumentDefinitions(): void {
+    if (!this.documentDefinition) {
+      return;
+    }
+    const {name, version} = this.documentDefinition.id;
     this.processDocumentDefinitionExists = {};
-    this.documentService
-      .findProcessDocumentDefinitions(this.documentDefinition?.id.name ?? '')
-      .subscribe((processDocumentDefinitions: ProcessDocumentDefinition[]) => {
-        processDocumentDefinitions.forEach(
-          (processDocumentDefinition: ProcessDocumentDefinition) => {
-            this.processDocumentDefinitionExists[
-              processDocumentDefinition.id.processDefinitionKey
-            ] = true;
-          }
-        );
-      });
+    this._subscriptions.add(
+      this.documentService
+        .findProcessDocumentDefinitionsByVersion(name, version)
```

Comment:
What's the point of adding this to _subscriptions? .findProcessDocumentDefinitionsByVersion is a http call that closes after one emission